src/mobu/handlers/external.py

```diff
@@ -232,3 +253,33 @@ async def get_summary(
     context: Annotated[RequestContext, Depends(context_dependency)],
 ) -> list[FlockSummary]:
     return context.manager.summarize_flocks()
+
+
+@external_router.post(
+    "/github/webhook",
+    summary="GitHub webhooks",
+    description="This endpoint receives webhook events from GitHub.",
+    status_code=202,
+)
+async def post_github_webhook(
+    context: Annotated[RequestContext, Depends(anonymous_context_dependency)],
+) -> Response:
+    """Process GitHub webhook events.
+
+    This should be exposed via a Gafaelfawr anonymous ingress.
+    """
+    webhook_secret = config.github_webhook_secret
+    body = await context.request.body()
+    event = Event.from_http(
+        context.request.headers, body, secret=webhook_secret
+    )
+
+    # Bind the X-GitHub-Delivery header to the logger context; this
+    # identifies the webhook request in GitHub's API and UI for
+    # diagnostics
+    logger = context.logger.bind(github_delivery=event.delivery_id)
```

Comment:
This change isn't picked up by the underlying service code, since loggers are immutable. I recommend instead adding a `rebind_logger` method on the context that pushes the rebound logger down to the factory, as in https://github.com/lsst-sqre/nublado/blob/main/controller/src/controller/dependencies/context.py#L65